// conventional-commits-detector/dist/lib/detectors.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _regex = require('./regex');

var _regex2 = _interopRequireDefault(_regex);

var _util = require('./util');

exports['default'] = {
  angular: function angular(commit) {
    var parts = commit.match(_regex2['default'].angular);

    if (!parts) {
      return false;
    }

    if ((0, _util.isLowerCase)(parts[0][0])) {
      return true;
    }
  },
  atom: function atom(commit) {
    var match = commit.match(_regex2['default'].atom);

    if (match) {
      return true;
    }

    return commit.match(/^Prepare (.*?) release$/);
  },
  ember: function ember(commit) {
    return commit.match(_regex2['default'].ember);
  },
  eslint: function eslint(commit) {
    var parts = commit.match(_regex2['default'].eslint);

    if (!parts) {
      return false;
    }

    if ((0, _util.isUpperCase)(parts[0][0])) {
      return true;
    }
  },
  jquery: function jquery(commit) {
    var parts = commit.match(_regex2['default'].jquery);

    if (!parts) {
      return false;
    }

    if ((0, _util.isUpperCase)(parts[0][0])) {
      return true;
    }
  },
  jshint: function jshint(commit) {
    return commit.match(_regex2['default'].jshint);
  }
};
module.exports = exports['default'];
